unitTests/test_Photoshop.cpp

```diff
@@ -0,0 +1,174 @@
+                                  0x49, 0x4d, 0x04, 0x25, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x3f, 0x65, 0x16, 0xda,
+                                  0x51, 0x3f, 0xfe, 0x5c, 0xbb, 0x52, 0xf3, 0x2e, 0x36, 0x7b, 0x97, 0x3d};
+
+  const Exiv2::byte* record;
+  uint32_t sizeHdr = 0;
+  uint32_t sizeData = 0;
+
+  ASSERT_EQ(0, Photoshop::locateIptcIrb(data.data(), data.size(), &record, &sizeHdr, &sizeData));
+  ASSERT_EQ(12, sizeHdr);
+  ASSERT_EQ(27, sizeData);
+}
+
+TEST(Photoshop_locateIptcIrb, returns3withoutIptcMarker) {
+  // Data taken from file test/data/DSC_3079.jpg
+  // The IPTC marker (0x04 0x04) was manipulated to 0x03 0x04
+  const std::array<byte, 68> data{0x38, 0x42, 0x49, 0x4d, 0x03, 0x04, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1b, 0x1c, 0x01,
+                                  0x5a, 0x00, 0x03, 0x1b, 0x25, 0x47, 0x1c, 0x02, 0x00, 0x00, 0x02, 0x00, 0x04, 0x1c,
+                                  0x02, 0x19, 0x00, 0x07, 0x41, 0x6d, 0x65, 0x72, 0x69, 0x63, 0x61, 0x00, 0x38, 0x42,
+                                  0x49, 0x4d, 0x04, 0x25, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x3f, 0x65, 0x16, 0xda,
+                                  0x51, 0x3f, 0xfe, 0x5c, 0xbb, 0x52, 0xf3, 0x2e, 0x36, 0x7b, 0x97, 0x3d};
+
+  const Exiv2::byte* record;
+  uint32_t sizeHdr = 0;
+  uint32_t sizeData = 0;
+
+  ASSERT_EQ(3, Photoshop::locateIptcIrb(data.data(), data.size(), &record, &sizeHdr, &sizeData));
+}
+
+TEST(Photoshop_locatePreviewIrb, returns0withGoodPreviewIrb) {
+  // Data taken from file test/data/DSC_3079.jpg
+  // The IPTC marker is 0x04 0x04 was transformed to a Preview one => (0x04 0x0c)
+  const std::array<byte, 68> data{0x38, 0x42, 0x49, 0x4d, 0x04, 0x0c, 0x00, 0x00, 0x00, 0x00, 0x00, 0x1b, 0x1c, 0x01,
+                                  0x5a, 0x00, 0x03, 0x1b, 0x25, 0x47, 0x1c, 0x02, 0x00, 0x00, 0x02, 0x00, 0x04, 0x1c,
+                                  0x02, 0x19, 0x00, 0x07, 0x41, 0x6d, 0x65, 0x72, 0x69, 0x63, 0x61, 0x00, 0x38, 0x42,
+                                  0x49, 0x4d, 0x04, 0x25, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10, 0x3f, 0x65, 0x16, 0xda,
+                                  0x51, 0x3f, 0xfe, 0x5c, 0xbb, 0x52, 0xf3, 0x2e, 0x36, 0x7b, 0x97, 0x3d};
+
+  const Exiv2::byte* record;
+  uint32_t sizeHdr = 0;
+  uint32_t sizeData = 0;
+
+  ASSERT_EQ(0, Photoshop::locatePreviewIrb(data.data(), data.size(), &record, &sizeHdr, &sizeData));
+  ASSERT_EQ(12, sizeHdr);
+  ASSERT_EQ(27, sizeData);
+}
+
+// --------------------------------
+
+TEST(Photoshop_setIptcIrb, withEmptyDataReturnsEmptyBuffer) {
+  const IptcData iptc;
+  DataBuf buf = Photoshop::setIptcIrb(nullptr, 0, iptc);
+  ASSERT_TRUE(buf.empty());
+}
+
+TEST(Photoshop_setIptcIrb, detectIntegerOverflow_withDataFromPOC2179) {
+  const std::array<byte, 141> data{
+      0x38, 0x42, 0x49, 0x4d, 0x20, 0x20, 0x00, 0x20, 0x00, 0x00, 0x00, 0x00, 0x38, 0x42, 0x49, 0x4d, 0x04, 0x04,
```

Comment:
This is the test which is replicating the case detected by oss-fuzz. Note that the assertion is checking that an exception is thrown.